src/NUnitFramework/tests/SolutionTests/NuspecDependenciesTests.cs

```diff
@@ -0,0 +1,174 @@
+// Copyright (c) Charlie Poole, Rob Prouse and Contributors. MIT License - see LICENSE.txt
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Xml.Linq;
+
+namespace NUnit.Framework.Tests.SolutionTests
+{
+    /// <summary>
+    /// Checks that the csproj of the framework projects are reflected correctly in the nuspec files.
+    /// </summary>
+    internal class NuspecDependenciesTests
+    {
+        private const string Root = "../../../../../../";
+
+        [Test]
+        public void NUnitFrameworkNuspecContainsCorrectDependencies()
+        {
+            var nuspec = new NuspecReader("framework/nunit.nuspec");
+            var csproj = new CsprojReader("framework/nunit.framework.csproj");
+            var nuspecPackages = nuspec.ExtractNuspecPackages();
+            var csprojPackages = csproj.ExtractCsprojPackages();
+            VerifyDependencies.ComparePackages(csprojPackages, nuspecPackages);
```

Comment:
This checks that the nuspec contains all packages from the csproj.
Should we also do a reverse check to ensure that if we drop a package from the csproj it is also dropped from the nuspec?
Also do we need to add a separate test to ensure that the nuspec contains entries for all supported TargetFramework?

Reply:
Yes, agree!